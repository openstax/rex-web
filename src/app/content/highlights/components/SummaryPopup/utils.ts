import { Highlight } from '@openstax/highlighter/dist/api';
import queryString from 'querystring';
import React from 'react';
import { useSelector } from 'react-redux';
import { systemQueryParameters } from '../../../../navigation/selectors';
import { Book } from '../../../types';
import { findArchiveTreeNodeById } from '../../../utils/archiveTreeUtils';
import { getBookPageUrlAndParams } from '../../../utils/urlUtils';

export const useCreateHighlightLink = ({ sourceId, id, anchor: elementId }: Highlight, book?: Book) => {
  const [link, setLink] = React.useState('');
  const systemQueryParams = useSelector(systemQueryParameters);

  React.useEffect(() => {
    const page = book ? findArchiveTreeNodeById(book.tree, sourceId) : undefined;
    const data: Record<string, string> = {
      ...systemQueryParams,
      target: JSON.stringify({ id, type: 'highlight' }),
    };
    const target = queryString.stringify(data) + `#${elementId}`;
    setLink(`${page && book ? getBookPageUrlAndParams(book, page).url : ''}?${target}`);
  }, [id, book, elementId, sourceId, systemQueryParams]);

  return link;
};
