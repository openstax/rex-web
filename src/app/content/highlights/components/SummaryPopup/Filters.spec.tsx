import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestServices from '../../../../../test/createTestServices';
import createTestStore from '../../../../../test/createTestStore';
import MessageProvider from '../../../../../test/MessageProvider';
import { book as archiveBook } from '../../../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../../../test/mocks/osWebLoader';
import Checkbox from '../../../../components/Checkbox';
import { DropdownToggle } from '../../../../components/Dropdown';
import * as Services from '../../../../context/Services';
import { MiddlewareAPI, Store } from '../../../../types';
import { formatBookData, stripIdVersion } from '../../../utils';
import { receiveHighlightsTotalCounts, updateSummaryFilters } from '../../actions';
import Filters from './Filters';

describe('Filters', () => {
  let store: Store;
  let services: ReturnType<typeof createTestServices> & MiddlewareAPI;
  let dispatch: jest.SpyInstance;
  const book = formatBookData(archiveBook, mockCmsBook);

  beforeEach(() => {
    store = createTestStore();
    services = {
      ...createTestServices(),
      dispatch: store.dispatch,
      getState: store.getState,
    };
    dispatch = jest.spyOn(store, 'dispatch');
  });

  it('matches snapshot and renders proper aria labels', () => {
    const pageId = stripIdVersion(book.tree.contents[0].id);
    store.dispatch(receiveHighlightsTotalCounts({
      [pageId]: {
        [HighlightColorEnum.Green]: 1,
        [HighlightColorEnum.Yellow]: 1,
        [HighlightColorEnum.Blue]: 1,
        [HighlightColorEnum.Pink]: 1,
        [HighlightColorEnum.Purple]: 1,
      },
    }, new Map()));

    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Filters />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    renderer.act(() => {
      const [chapterFilterToggle, colorFilterToggle] = component.root.findAllByType(DropdownToggle);
      chapterFilterToggle.props.onClick();
      colorFilterToggle.props.onClick();
    });

    const labelBlueKey = component.root.findAllByProps({ id: 'i18n:highlighting:colors:blue' });
    const labelGreenKey = component.root.findAllByProps({ id: 'i18n:highlighting:colors:green' });
    const labelPurpleKey = component.root.findAllByProps({ id: 'i18n:highlighting:colors:purple' });
    const labelYellowKey = component.root.findAllByProps({ id: 'i18n:highlighting:colors:yellow' });
    const labelPinkKey = component.root.findAllByProps({ id: 'i18n:highlighting:colors:pink' });

    // There should be 2 elements with each of these ids. One is ColorLabel and second is FiltersListColor
    expect(labelBlueKey.length).toEqual(2);
    expect(labelGreenKey.length).toEqual(2);
    expect(labelPurpleKey.length).toEqual(2);
    expect(labelYellowKey.length).toEqual(2);
    expect(labelPinkKey.length).toEqual(2);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispatches updateSummaryFilters action', () => {
    const pageId = stripIdVersion(book.tree.contents[0].id);
    store.dispatch(receiveHighlightsTotalCounts({
      [pageId]: {
        [HighlightColorEnum.Green]: 1,
        [HighlightColorEnum.Yellow]: 1,
        [HighlightColorEnum.Blue]: 1,
        [HighlightColorEnum.Pink]: 1,
        [HighlightColorEnum.Purple]: 1,
      },
    }, new Map()));

    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Filters />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    renderer.act(() => {
      const [, colorFilterToggle] = component.root.findAllByType(DropdownToggle);
      colorFilterToggle.props.onClick();
    });

    const [yellowCheckbox] = component.root.findAllByType(Checkbox);

    renderer.act(() => {
      yellowCheckbox.props.onChange();
    });

    expect(dispatch).toHaveBeenCalledWith(updateSummaryFilters({
      colors: { new: [], remove: [HighlightColorEnum.Yellow] },
    }));

    renderer.act(() => {
      yellowCheckbox.props.onChange();
    });

    expect(dispatch).toHaveBeenCalledWith(updateSummaryFilters({
      colors: { new: [HighlightColorEnum.Yellow], remove: [] },
    }));
  });
});
