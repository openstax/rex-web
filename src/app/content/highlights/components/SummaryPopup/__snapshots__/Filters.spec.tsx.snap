// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters matches snapshot and renders proper aria labels 1`] = `
<div
  className="Filters-sc-1fn81fx-3 gZlQQL"
>
  <div
    className="Filters__FiltersTopBar-sc-1fn81fx-2 fWMshY"
  >
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter highlights by Chapter"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 gsKueT Button__PlainButton-ayg7nk-2 chgswm"
        data-analytics-label="Filter highlights by Chapter"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Chapter
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="up"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <div
        className="ChapterFilter-tcdlu-7 iaQiZB"
        tabIndex={-1}
      >
        <div
          className="AllOrNone-sc-7m1y94-0 divvKQ"
        >
          <button
            className="Button__PlainButton-ayg7nk-2 Button__ButtonLink-ayg7nk-3 flLYdq"
            onClick={[Function]}
          >
            All
          </button>
          <span>
            |
          </span>
          <button
            className="Button__PlainButton-ayg7nk-2 Button__ButtonLink-ayg7nk-3 flLYdq"
            onClick={[Function]}
          >
            None
          </button>
        </div>
        <div
          className="ChapterFilter__Row-tcdlu-0 nqIYu"
        />
      </div>
    </div>
    <div
      className="Dropdown-rmc6yw-5 fBPFfC Dropdown__TabHiddenDropDown-rmc6yw-1 kbMrSv"
    >
      <button
        aria-label="Filter highlights by Color"
        className="Dropdown__DropdownToggle-rmc6yw-0 gpiKKF Filters__Toggle-sc-1fn81fx-1 gsKueT Button__PlainButton-ayg7nk-2 chgswm"
        data-analytics-label="Filter highlights by Color"
        onClick={[Function]}
      >
        <div
          tabIndex={-1}
        >
          Color
          <svg
            aria-hidden="true"
            className="Filters__AngleIcon-sc-1fn81fx-0 jKNYyQ StyledIconBase-sc-bdy9j4 cCvWXa"
            direction="up"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 320 512"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <div
        className="ColorFilter-sc-1ic50x3-1 gDnnAT"
        tabIndex={-1}
      >
        <div
          className="AllOrNone-sc-7m1y94-0 divvKQ"
        >
          <button
            className="Button__PlainButton-ayg7nk-2 Button__ButtonLink-ayg7nk-3 flLYdq"
            onClick={[Function]}
          >
            All
          </button>
          <span>
            |
          </span>
          <button
            className="Button__PlainButton-ayg7nk-2 Button__ButtonLink-ayg7nk-3 flLYdq"
            onClick={[Function]}
          >
            None
          </button>
        </div>
        <label
          className="Checkbox-f0dje9-2 ksczzL"
        >
          <input
            checked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="Checkbox__CustomCheckbox-f0dje9-1 gLNaiW"
          >
            <svg
              aria-hidden="true"
              className="Checkbox__CheckIcon-f0dje9-0 irsTpS StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="ColorIndicator-sc-1w5zl7g-2 hxgXOu"
          >
            <svg
              aria-hidden="true"
              className="ColorIndicator__CheckIcon-sc-1w5zl7g-0 Lufbl StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
            <span
              className="ColorIndicator__FocusedStyle-sc-1w5zl7g-1 kwsovu"
              size="small"
              style={
                Object {
                  "focused": "#fed200",
                  "label": "yellow",
                  "passive": "#ffff8a",
                }
              }
            />
          </div>
          <span
            className="ColorFilter__ColorLabel-sc-1ic50x3-0 iOVncN"
          >
            yellow
          </span>
        </label>
        <label
          className="Checkbox-f0dje9-2 ksczzL"
        >
          <input
            checked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="Checkbox__CustomCheckbox-f0dje9-1 gLNaiW"
          >
            <svg
              aria-hidden="true"
              className="Checkbox__CheckIcon-f0dje9-0 irsTpS StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="ColorIndicator-sc-1w5zl7g-2 kUYWjn"
          >
            <svg
              aria-hidden="true"
              className="ColorIndicator__CheckIcon-sc-1w5zl7g-0 Lufbl StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
            <span
              className="ColorIndicator__FocusedStyle-sc-1w5zl7g-1 fqiTys"
              size="small"
              style={
                Object {
                  "focused": "#92d101",
                  "label": "green",
                  "passive": "#def99f",
                }
              }
            />
          </div>
          <span
            className="ColorFilter__ColorLabel-sc-1ic50x3-0 iOVncN"
          >
            green
          </span>
        </label>
        <label
          className="Checkbox-f0dje9-2 ksczzL"
        >
          <input
            checked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="Checkbox__CustomCheckbox-f0dje9-1 gLNaiW"
          >
            <svg
              aria-hidden="true"
              className="Checkbox__CheckIcon-f0dje9-0 irsTpS StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="ColorIndicator-sc-1w5zl7g-2 fkmJHS"
          >
            <svg
              aria-hidden="true"
              className="ColorIndicator__CheckIcon-sc-1w5zl7g-0 Lufbl StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
            <span
              className="ColorIndicator__FocusedStyle-sc-1w5zl7g-1 eHIgYD"
              size="small"
              style={
                Object {
                  "focused": "#00c3ed",
                  "label": "blue",
                  "passive": "#c8f5ff",
                }
              }
            />
          </div>
          <span
            className="ColorFilter__ColorLabel-sc-1ic50x3-0 iOVncN"
          >
            blue
          </span>
        </label>
        <label
          className="Checkbox-f0dje9-2 ksczzL"
        >
          <input
            checked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="Checkbox__CustomCheckbox-f0dje9-1 gLNaiW"
          >
            <svg
              aria-hidden="true"
              className="Checkbox__CheckIcon-f0dje9-0 irsTpS StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="ColorIndicator-sc-1w5zl7g-2 gdbVRC"
          >
            <svg
              aria-hidden="true"
              className="ColorIndicator__CheckIcon-sc-1w5zl7g-0 Lufbl StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
            <span
              className="ColorIndicator__FocusedStyle-sc-1w5zl7g-1 AHChX"
              size="small"
              style={
                Object {
                  "focused": "#545ec8",
                  "label": "purple",
                  "passive": "#cbcfff",
                }
              }
            />
          </div>
          <span
            className="ColorFilter__ColorLabel-sc-1ic50x3-0 iOVncN"
          >
            purple
          </span>
        </label>
        <label
          className="Checkbox-f0dje9-2 ksczzL"
        >
          <input
            checked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="Checkbox__CustomCheckbox-f0dje9-1 gLNaiW"
          >
            <svg
              aria-hidden="true"
              className="Checkbox__CheckIcon-f0dje9-0 irsTpS StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="ColorIndicator-sc-1w5zl7g-2 bhqGqa"
          >
            <svg
              aria-hidden="true"
              className="ColorIndicator__CheckIcon-sc-1w5zl7g-0 Lufbl StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
              />
            </svg>
            <span
              className="ColorIndicator__FocusedStyle-sc-1w5zl7g-1 cxneHl"
              size="small"
              style={
                Object {
                  "focused": "#de017e",
                  "label": "pink",
                  "passive": "#ffc5e1",
                }
              }
            />
          </div>
          <span
            className="ColorFilter__ColorLabel-sc-1ic50x3-0 iOVncN"
          >
            pink
          </span>
        </label>
      </div>
    </div>
    <button
      aria-label="print"
      className="styled__PlainButton-sc-1119auh-0 PrintButton-sc-1bdptk0-0 iIYjST styled__PrintOptWrapper-sc-1119auh-1 kbtaBS"
      data-analytics-label="print"
      data-testid="print"
      disabled={false}
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="styled__PrintIcon-sc-1119auh-3 iKTgDK StyledIconBase-sc-bdy9j4 cCvWXa"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 512 512"
      >
        <path
          d="M448 192V77.25c0-8.49-3.37-16.62-9.37-22.63L393.37 9.37c-6-6-14.14-9.37-22.63-9.37H96C78.33 0 64 14.33 64 32v160c-35.35 0-64 28.65-64 64v112c0 8.84 7.16 16 16 16h48v96c0 17.67 14.33 32 32 32h320c17.67 0 32-14.33 32-32v-96h48c8.84 0 16-7.16 16-16V256c0-35.35-28.65-64-64-64zm-64 256H128v-96h256v96zm0-224H128V64h192v48c0 8.84 7.16 16 16 16h48v96zm48 72c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
          fill="currentColor"
        />
      </svg>
      <span
        className="styled__PrintOptions-sc-1119auh-2 bRCGrc"
      >
        Print
      </span>
    </button>
  </div>
  <ul
    className="FiltersList-sc-1sdnk9e-3 fQoHdD"
  >
    <li
      className="FiltersList__FilterListItem-sc-1sdnk9e-2 hhDKAf"
    >
      <button
        aria-label="Remove breadcrumb for color blue"
        className="Button__PlainButton-ayg7nk-2 FiltersList__StyledPlainButton-sc-1sdnk9e-0 gUsiXX"
        data-analytics-label="Remove breadcrumb for color blue"
        onClick={[Function]}
      >
        <svg
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
      <span
        className="FiltersList__ItemLabel-sc-1sdnk9e-1 TUwad"
      >
        blue
      </span>
    </li>
    <li
      className="FiltersList__FilterListItem-sc-1sdnk9e-2 hhDKAf"
    >
      <button
        aria-label="Remove breadcrumb for color green"
        className="Button__PlainButton-ayg7nk-2 FiltersList__StyledPlainButton-sc-1sdnk9e-0 gUsiXX"
        data-analytics-label="Remove breadcrumb for color green"
        onClick={[Function]}
      >
        <svg
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
      <span
        className="FiltersList__ItemLabel-sc-1sdnk9e-1 TUwad"
      >
        green
      </span>
    </li>
    <li
      className="FiltersList__FilterListItem-sc-1sdnk9e-2 hhDKAf"
    >
      <button
        aria-label="Remove breadcrumb for color pink"
        className="Button__PlainButton-ayg7nk-2 FiltersList__StyledPlainButton-sc-1sdnk9e-0 gUsiXX"
        data-analytics-label="Remove breadcrumb for color pink"
        onClick={[Function]}
      >
        <svg
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
      <span
        className="FiltersList__ItemLabel-sc-1sdnk9e-1 TUwad"
      >
        pink
      </span>
    </li>
    <li
      className="FiltersList__FilterListItem-sc-1sdnk9e-2 hhDKAf"
    >
      <button
        aria-label="Remove breadcrumb for color purple"
        className="Button__PlainButton-ayg7nk-2 FiltersList__StyledPlainButton-sc-1sdnk9e-0 gUsiXX"
        data-analytics-label="Remove breadcrumb for color purple"
        onClick={[Function]}
      >
        <svg
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
      <span
        className="FiltersList__ItemLabel-sc-1sdnk9e-1 TUwad"
      >
        purple
      </span>
    </li>
    <li
      className="FiltersList__FilterListItem-sc-1sdnk9e-2 hhDKAf"
    >
      <button
        aria-label="Remove breadcrumb for color yellow"
        className="Button__PlainButton-ayg7nk-2 FiltersList__StyledPlainButton-sc-1sdnk9e-0 gUsiXX"
        data-analytics-label="Remove breadcrumb for color yellow"
        onClick={[Function]}
      >
        <svg
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
      <span
        className="FiltersList__ItemLabel-sc-1sdnk9e-1 TUwad"
      >
        yellow
      </span>
    </li>
  </ul>
</div>
`;
